crates/polars-plan/src/dsl/function_expr/ewm.rs

```diff
@@ -4,6 +4,43 @@ pub(super) fn ewm_mean(s: &Series, options: EWMOptions) -> PolarsResult<Series>
     polars_ops::prelude::ewm_mean(s, options)
 }
 
+pub(super) fn ewm_mean_by(
+    s: &[Series],
+    half_life: Duration,
+    warn_if_unsorted: bool,
+) -> PolarsResult<Series> {
+    let time_zone = match s[1].dtype() {
+        DataType::Datetime(_, Some(time_zone)) => Some(time_zone.as_str()),
+        _ => None,
+    };
+    if half_life.negative() {
+        polars_bail!(InvalidOperation: "half_life cannot be negative");
+    }
+    if !half_life.is_constant_duration(time_zone) {
+        polars_bail!(InvalidOperation: "expected `half_life` to be a constant duration \
+        (i.e. one independent of differing month durations or of daylight savings time), got {}.\n\n
+        You may want to try:\n\
+        - using `'730h'` instead of `'1mo'`;\n\
+        - using `'24h'` instead of `'1d'` if your series is time-zone-aware;
+        ", half_life);
+    }
+    if s[1].is_sorted_flag() != IsSorted::Ascending && warn_if_unsorted {
+        polars_warn!(format!(
+            "Series is not known to be sorted by `by` column in {} operation.\n\
+            \n\
+            To silence this warning, you may want to try:\n\
+            - sorting your data by your `by` column beforehand;\n\
+            - setting `.set_sorted()` if you already know your data is sorted;\n\
+            - passing `warn_if_unsorted=False` if this warning is a false-positive\n  \
+                (this is known to happen when combining rolling aggregations with `over`);\n\n\
+            before passing calling the rolling aggregation function.\n",
+            "ewm_mean_by",
+        ));
+    }
```

Comment:
after this, I plan to try addressing https://github.com/pola-rs/polars/pull/11134/files again...hopefully my general understanding has improved and I'll be able to get it done this time